UKHO.D365CallbackDistributorStub.API/UKHO.D365CallbackDistributorStub.API/Services/DistributionService.cs

```diff
@@ -41,5 +53,55 @@ public static bool SaveDistributorRequest(CustomCloudEvent cloudEvent)
                 return s_recordDistributorRequestQueue.ToList();
             }
         }
+
+        public bool SaveCommandDistributorRequest(string subject, HttpStatusCode? statusCode)
+        {
+            try
+            {
+                CommandDistributionRequest? commandDistributorRequest = s_CommandDistributionList.Where(a => a.Subject == subject).LastOrDefault();
+                if (commandDistributorRequest != null)
+                {
+                    if (statusCode == null)
+                    {
+                        s_CommandDistributionList.Remove(commandDistributorRequest);
+                    }
+                    else
+                    {
+                        commandDistributorRequest.HttpStatusCode = statusCode ?? Ok;
+                    }
+                }
+                else
+                {
+                    if (statusCode != null)
+                    {
+                        s_CommandDistributionList.Add(new CommandDistributionRequest
+                        {
+                            Subject = subject,
+                            HttpStatusCode = statusCode ?? Ok
+                        });
+                    }
+                    else
+                    {
+                        _logger.LogInformation("Request not found in memory for subject: {subject}", subject);
+                        return true;
+                    }
+                }
+
+                if (s_CommandDistributionList.Count >= 50)
+                {
+                    s_CommandDistributionList.RemoveAt(0);
+                }
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        public CommandDistributionRequest? SubjectInCommandDistributionList(string? subject)
+        {
+            return s_CommandDistributionList.Where(a => a.Subject == subject).LastOrDefault();
```

Comment:
Please simplify LINQ expression as below:
```suggestion
            return s_CommandDistributionList.LastOrDefault(a => a.Subject == subject);
```

Reply:
Fixed